common/beerocks/bcl/source/network/network_utils.cpp

```diff
@@ -363,6 +365,95 @@ static int parseRoutes(struct nlmsghdr *nlHdr, std::shared_ptr<route_info> rtInf
     return (0);
 }
 
+std::shared_ptr<std::unordered_map<std::string, std::string>>
+network_utils::get_arp_table(bool mac_as_key)
+{
+    auto arp_table_ptr = std::make_shared<std::unordered_map<std::string, std::string>>();
+    auto &arp_table    = *arp_table_ptr.get();
+
+    // Allocate a new netlink socket
+    auto nl_socket = nl_socket_alloc();
+    if (nl_socket == nullptr) {
+        LOG(ERROR) << "Failed allocating netlink socket!";
+        return nullptr;
+    }
+
+    int err = nl_connect(nl_socket, NETLINK_ROUTE);
+    if (err != 0) {
+        LOG(ERROR) << "Failed connecting the netlink socket: " << nl_geterror(err);
+        nl_socket_free(nl_socket);
+        return nullptr;
+    }
+
+    // ARP table cache
+    struct nl_cache *neighbor_table_cache = nullptr;
+
+    // Allocate the cache and fill it with data
+    err = rtnl_neigh_alloc_cache(nl_socket, &neighbor_table_cache);
+    if (err != 0) {
+        LOG(ERROR) << "Failed probing ARP table: " << nl_geterror(err);
+        nl_socket_free(nl_socket);
+        return nullptr;
+    }
+
+    nl_socket_free(nl_socket);
+
+    auto enteries_count = nl_cache_nitems(neighbor_table_cache);
+
+    constexpr const char *error_message = "neighbor_entry is null";
+    auto neighbor_entry = reinterpret_cast<rtnl_neigh *>(nl_cache_get_first(neighbor_table_cache));
+    if (!neighbor_entry) {
+        LOG(ERROR) << error_message;
+        nl_cache_free(neighbor_table_cache);
+        return nullptr;
+    }
+
+    char buffer_mac[MAC_ADDR_CHAR_SIZE + 1]; // +1 for null terminator
+    char buffer_ip[IPV4_ADDR_CHAR_SIZE + 1]; // +1 for null terminator
+
+    for (int i = 0; i < enteries_count; i++) {
```

Comment:
You don't need a for-loop to iterate over the cache.
`nl_cache_get_next` returns NULL at the end of the list.

Reply:
It is a good idea, that way I remove the entry check on the loop body, so I have only one place to print the error, and I could get rid of the constexpr error.